src/sqlfluff/rules/L003.py

```diff
@@ -776,136 +665,381 @@ def _process_current_line(
                             ),
                         ],
                     )
-                elif this_indent_num > comp_indent_num + indent_diff:
-                    # Calculate the lowest ok indent:
-                    desired_indent = self._make_indent(
-                        num=comp_indent_num - this_indent_num,
-                        indent_unit=self.indent_unit,
-                        tab_space_size=self.tab_space_size,
-                    )
+            elif this_indent_num < comp_indent_num:
+                memory.problem_lines.add(this_line_no)
+                return LintResult(
+                    anchor=trigger_segment,
+                    memory=memory,
+                    description=_Desc(
+                        expected=this_indent_num + 1,
+                        found=this_indent_num,
+                        compared_to=prev_line.line_no,
+                    ),
+                    fixes=[
+                        LintFix.create_before(
+                            trigger_segment,
+                            [
+                                WhitespaceSegment(
+                                    # Make the minimum indent for it to be ok.
+                                    raw=self._make_indent(
+                                        num=comp_indent_num - this_indent_num,
```

Comment:
why cant we coerce to comp_intent_num instead of add 1 worth of indent element. (tests dont pass if we do this)